Tests/FluentAssertions.Specs/Specialized/TaskOfTAssertionSpecs.cs

```diff
@@ -129,5 +132,205 @@ public void When_task_completes_slow_async_it_should_fail()
+
+            var asyncObject = new AsyncClass();
+            Func<Task<int>> someFunc = () => asyncObject.ReturnTaskInt();
+
+            // Act
+            Func<Task> act = () => someFunc.Should().NotThrowAfterAsync(waitTime, pollInterval);
+
+            // Assert
+            act.Should().Throw<ArgumentOutOfRangeException>()
+                .WithMessage("* value of waitTime must be non-negative*");
+        }
+
+        [Fact]
+        public void When_poll_interval_is_negative_for_async_func_it_should_throw()
+        {
+            // Arrange
+            var waitTime = 10.Milliseconds();
+            var pollInterval = -1.Milliseconds();
+
+            var asyncObject = new AsyncClass();
+            Func<Task<int>> someFunc = () => asyncObject.ReturnTaskInt();
+
+            // Act
+            Func<Task> act = () => someFunc.Should().NotThrowAfterAsync(waitTime, pollInterval);
+
+            // Assert
+            act.Should().Throw<ArgumentOutOfRangeException>()
+                .WithMessage("* value of pollInterval must be non-negative*");
+        }
+
+        [Fact]
+        public void When_no_exception_should_be_thrown_for_null_async_func_after_wait_time_it_should_throw()
+        {
+            // Arrange
+            var waitTime = 2.Seconds();
+            var pollInterval = 10.Milliseconds();
+
+            Func<Task<int>> func = null;
+
+            // Act
+            Func<Task> action = () => func.Should()
+                .NotThrowAfterAsync(waitTime, pollInterval, "we passed valid arguments");
+
+            // Assert
+            action.Should().Throw<XunitException>()
+                .WithMessage("*but found <null>*");
+        }
+
+        [Fact]
+        public void When_no_exception_should_be_thrown_for_async_func_after_wait_time_but_it_was_it_should_throw()
+        {
+            // Arrange
+            var waitTime = 2.Seconds();
+            var pollInterval = 10.Milliseconds();
+
+            var clock = new FakeClock();
+            var timer = clock.StartTimer();
+            clock.CompleteAfter(waitTime);
+
+            Func<Task<int>> throwLongerThanWaitTime = async () =>
+            {
+                if (timer.Elapsed <= waitTime.Multiply(1.5))
+                {
+                    throw new ArgumentException("An exception was forced");
+                }
+
+                await Task.Yield();
+                return 42;
+            };
+
+            // Act
+            Func<Task> action = () => throwLongerThanWaitTime.Should(clock)
+                .NotThrowAfterAsync(waitTime, pollInterval, "we passed valid arguments");
+
+            // Assert
+            action.Should().Throw<XunitException>()
+                .WithMessage("Did not expect any exceptions after 2s because we passed valid arguments*");
+        }
+
+        [Fact]
+        public void When_no_exception_should_be_thrown_for_async_func_after_wait_time_and_none_was_it_should_not_throw()
```

Comment:
I would skip the `async_func` part since that should be clear from the context.

Reply:
fixed